tensorflow/compiler/mlir/tensorflow/ir/tf_generated_ops.td

```diff
@@ -4434,7 +4434,7 @@ dimension of size 1 added.}]>:$output
   ];
 }
 
-def TF_Expm1Op : TF_Op<"Expm1", [NoSideEffect, TF_SameOperandsAndResultTypeResolveRef]> {
+def TF_Expm1Op : TF_Op<"Expm1", [NoSideEffect, SameOperandsAndResultType]> {
```

Comment:
why is this change from TF_SameOperandsAndResultTypeResolveRef to SameOperandsAndResultType?

Reply:
``TF_SameOperandsAndResultTypeResolveRef`` is somehow not accepting the pattern ``TF_Expm1Op $features``. It is failing the build for this.

Reply:
Ah, I see.

Rewrite patterns have better support for SameOperandsAndResultType trait for ops that don't have shape inference function. Declaring shape inference function for this op should make it work.

Reply:
Seems like this is still outdated?

Reply:
yes I am working on this. Will soon update this.

Reply:
@smit-hinsu can you elaborate on shape inference function? Are you referring to ``InferShapedTypeOpInterface``?

Reply:
@smit-hinsu Added the shape inference function. Please re-review. Thanks!